src/Database/LSMTree/Internal.hs

```diff
@@ -541,13 +548,37 @@ updates ::
   -> m ()
 updates es th = withOpenTable th $ \thEnv -> do
     -- A placeholder implementation that is sufficient to pass the tests, but
-    -- keeps all entries in memory.
-    -- TODO: flush write buffer when full
+    -- simply writes small runs to disk without merging.
     -- TODO: merge runs when level becomes full
-    modifyMVar_ (tableWriteBuffer thEnv) $ \wb -> do
-      return $ foldl' (flip (uncurry (WB.addEntry resolve))) wb es
-  where
-    resolve = resolveMupsert (tableConfig th)
+    modifyMVar_ (tableContent thEnv) $ \tableContent -> do
+      let !wb = WB.addEntries (resolveMupsert (tableConfig th)) es $
+                  tableWriteBuffer tableContent
+      if WB.sizeInBytes wb <= fromIntegral (confWriteBufferAlloc (tableConfig th))
```

Comment:
We might actually just want to look at number of entries for now.

For the runs, this simplifies things quite a bit (e.g. with mupsert a merged run might end up larger than the sum of its inputs), so it would make sense to do it for the write buffer as well.

Reply:
I think it would be useful to still keep the size calculation in here

Reply:
I think on balance we should drop it (but keep the code on a branch in case we decide to resurrect it).

Reply:
Yes, I think we can keep this branch. I've opened #259 with the alternative approach based on number of entries